controllers/rollingupgrade_controller.go

```diff
@@ -602,7 +610,7 @@ func (r *RollingUpgradeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 			logr.Info("Sync map with invalid entry for ", "name", ruObj.Name)
 		}
 	} else {
-		go r.Process(&ctx, ruObj)
+		r.Process(&ctx, ruObj)
```

Comment:
calling `r.Process()` as a goroutine bypasses the native mechanism in controller-runtime for `MaxConcurrentReconciles`, it is now blocking and `Reconcile()` is called asynchronously according to worker count (`--max-parallel`)

Reply:
Only downside here is that whomever was used to `infinite` worker count as the behavior (due to calling `r.Process()` as a goroutine), is now getting limited to whatever default we set (in this case 10) as max parallel reconciles.